In [1]:
import pandas as pd
import numpy as np

In [2]:
probes = pd.read_csv(r"F:\vbn_s3_cache\visual-behavior-neuropixels-0.5.0\project_metadata\probes.csv")

In [6]:
probe_areas = probes.structure_acronyms.values
# get entries that contains VISp, not VISpm
index = [i for i, areas in enumerate(probe_areas) if ("VISp" in areas and "VISpm" not in areas)]
len(index)

151

In [7]:
names = probes.name.values
print(np.unique(names[index]))

['probeC']


> So VISp is only in probeC

In [8]:
import os
# find all the sessions
sessions = os.listdir(r"F:\behavior_ecephys_sessions")
print(sessions)

['1044385384', '1044594870', '1047969464', '1047977240', '1048189115', '1048196054', '1049273528', '1049514117', '1051155866', '1052342277', '1052533639', '1053709239', '1053718935', '1053925378', '1053941483', '1055221968', '1055240613', '1055403683', '1055415082', '1062755779', '1063010385', '1064400234', '1064415305', '1064639378', '1064644573', '1065437523', '1065449881', '1065905010', '1065908084', '1067588044', '1067781390', '1069461581', '1071300149', '1079018673', '1081079981', '1081090969', '1081429294', '1081431006', '1086200042', '1086410738', '1087720624', '1087992708', '1089296550', '1090800639', '1090803859', '1091039376', '1091039902', '1092283837', '1092466205', '1093638203', '1093642839', '1093864136', '1093867806', '1095138995', '1095340643', '1096620314', '1096935816', '1098119201', '1099598937', '1099869737', '1104052767', '1104058216', '1104289498', '1104297538', '1105543760', '1105798776', '1108334384', '1108335514', '1108528422', '1108531612', '1109680280', '1109

In [12]:
session_ids = probes.ecephys_session_id.values[index]
# make sure all the sessions are in the list
for session in sessions:
    if int(session) not in session_ids:
        print(f"Session {session} not found in the list of sessions.")

Session 1109680280 not found in the list of sessions.
Session 1123100019 not found in the list of sessions.


In [58]:
# find all the sesions in 0.4.0
session_table = pd.read_csv(r"E:\jerry\vbn_s3_cache\visual-behavior-neuropixels-0.4.0\project_metadata\ecephys_sessions.csv")
session_ids_4 = session_table.ecephys_session_id.values
session_abnormal_hist = session_table.abnormal_histology.values
session_abnormal_activity = session_table.abnormal_activity.values
filter_abnormal_hist = [i for i in range(len(session_abnormal_hist)) if (type(session_abnormal_hist[i]) is float) and (type(session_abnormal_activity[i]) is float)]

In [59]:
len(filter_abnormal_hist)

103

In [60]:
session_ids_4 = session_ids_4[filter_abnormal_hist]
print(len(session_ids_4))

103


In [66]:
# find which session in 0.5.0 is not in 0.4.0
for session in sessions:
    if int(session) not in session_ids_4:
        print(f"{session}")

1079018673


> 1079018673 is not in 0.4.0 allensdk, removed to prevent problems

In [70]:
from pathlib import Path
import shutil

In [82]:
source_dir = Path(r"z:\Buzsakilabspace\LabShare\NoamNitzan\Open_Access\Allen_2022\data")
sessions = os.listdir(r"F:\behavior_ecephys_sessions")
for session in sessions:
    session_path = source_dir / f'session_{session}'
    if int(session) in [1069461581, 1099598937, 1099869737, 1104289498, 1109889304]: continue 
    if session_path.exists():
        print(f"{session}")
        lfp_path_src = source_dir / f'session_{session}' / 'probeC_lfp.nwb'
        lfp_path_dst = Path(f"F:/behavior_ecephys_sessions/{session}/probe_probeC_lfp.nwb")
        if not lfp_path_src.exists():
            print(f"LFP file for session {session} does not exist at the source path.")
            continue
        if not lfp_path_dst.exists():
            shutil.copy(lfp_path_src, lfp_path_dst)
        else:
            print(f"LFP file already exists at the dest path.")
    else:
        print(f"Session {session} not found in source directory.")

1044385384
LFP file already exists at the dest path.
1044594870
LFP file already exists at the dest path.
1047969464
LFP file already exists at the dest path.
1047977240
LFP file already exists at the dest path.
1048189115
LFP file already exists at the dest path.
1048196054
LFP file already exists at the dest path.
1049273528
LFP file already exists at the dest path.
1049514117
LFP file already exists at the dest path.
1051155866
LFP file already exists at the dest path.
1052342277
LFP file already exists at the dest path.
1052533639
LFP file already exists at the dest path.
1053709239
LFP file already exists at the dest path.
1053718935
LFP file already exists at the dest path.
1053925378
LFP file already exists at the dest path.
1053941483
LFP file already exists at the dest path.
1055221968
LFP file already exists at the dest path.
1055240613
LFP file already exists at the dest path.
1055403683
LFP file already exists at the dest path.
1055415082
LFP file already exists at the dest

> sessions without lfp:
<br>
> 1069461581, 1099598937, 1104289498, 1109889304
<br>
> 1086200042, 1086410738, 1093638203, 1093864136, 1093867806, 1095138995, 1095340643, 1109680280, 1123100019

In [ ]:
# for the above sessions, mount the allensdk to the drive and download the LFP files manually
# remember to change the name of the nwb files to probe_probeC_lfp.nwb
from allensdk.brain_observatory.behavior.behavior_project_cache import VisualBehaviorNeuropixelsProjectCache
cache = VisualBehaviorNeuropixelsProjectCache.from_s3_cache(r'F:\vbn_s3_cache')
session_ids = [1069461581, 1099598937, 1104289498, 1109889304, 
               
               1086200042, 1086410738, 1093638203, 1093864136, 
               1093867806, 1095138995, 1095340643, 1109680280, 1123100019]


In [91]:
for idx, session_id in enumerate(session_ids):
    print(session_id)
    if idx <= 11: continue
    session = cache.get_ecephys_session(session_id)
    session_probes = probes[probes.ecephys_session_id == session_id]
    session_probeC = session_probes[session_probes.name == 'probeC']
    session_probeC_id = session_probeC.ecephys_probe_id.values[0]
    session.get_lfp(session_probeC_id)
    

1069461581
1099598937
1104289498
1109889304
1086200042
1086410738
1093638203
1093864136
1093867806
1095138995
1095340643
1109680280
1123100019


IndexError: index 0 is out of bounds for axis 0 with size 0

> sessions that just don't have probeC:
<br>
> 1086200042, 1086410738, 1093638203, 1093864136, 1093867806, 1095138995, 1095340643, 1109680280, 1123100019

In [3]:
from scipy.sparse import coo_matrix, save_npz
import numpy as np
import shutil

In [4]:
from IPython.display import clear_output
for i in range(20):
    for j in range(9):
        clear_output(wait=True)
        print(f"Processing channel {i}, band {j}")
        src = rf"Z:\Buzsakilabspace\LabShare\NoamNitzan\Open_Access\Allen_2022\models\1044385384\attrs\attribution_scores_transformer_chan{i}_band{j}.npy"
        dst = f"F:/transformer_attrs/attribution_scores_transformer_chan{i}_band{j}.npz"
        data = np.load(src, allow_pickle=True)
        coo = coo_matrix(data)
        save_npz(dst, coo, compressed=True)

Processing channel 19, band 8
